core/state/node/transitions.go

```diff
@@ -507,6 +474,18 @@ type AddReverseProxyRuleTransition struct {
 	Rule *ReverseProxyRule `json:"rule"`
 }
 
+func GenAddReverseProxyRuleTransition(t ReverseProxyRuleType, matcher string, target string, appID string) *AddReverseProxyRuleTransition {
```

Comment:
In a follow up I would like to make all of these return `Transition` and the transitions themselves private types.